### ISSUE TITLE
Improve discoverability of the editor keyboard shortcuts

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58010]
## What problem does this address?

Right now, the editor keyboard shortcuts aren't that discoverable. A list of all the shortcuts is available a little buried down within the Options menu > Keyboard shortcuts.

Some buttons in the user interface do visually expose their associated keyboard shortcut (if any) in their tooltip but that comes with some downside. For example, buttons that show visible text _and_ also have an associated keyboard shortcut still show a tooltip even though most of the tooltip text just repeats the button visible text. We do that because the tooltip is the only way to visually expose the keyboard shortcut but that's not ideal. This is even more redundant when the 'Show button text labels' preference is enabled, as all the buttons with a shortcut will shoe repeated text in their tooltip. Example screenshots:

![Screenshot 2024-01-19 at 15 44 48](https://github.com/WordPress/gutenberg/assets/1682452/0c963883-e918-47ed-b510-39edc46a11f2)

![Screenshot 2024-01-19 at 16 08 01](https://github.com/WordPress/gutenberg/assets/1682452/d433c32a-5907-47fc-b6b2-fa16c8407494)

I would love to explore ideas to make the keyboard shortcuts better discoverable.

## What is your proposed solution?

1
Consider to add a visually hidden button that gets visible on focus at the very beginning of the DOM in the editor. Activating the button opens the existing Keyboard Shortcuts modal dialog. This button would be available as the very first thing in the editor for all keyboard users, including sighted users.

Other widely used web applications do the same thing. For example, Gmail does have a few visually hidden controls right after the Skip link. One of them opens the keyboard shortcuts list. The only difference is that these Gmail links are always visually hidden, while we'd want to reveal them on focus for all users. Screenshot from Gmail: I altered the CSS to make these links visible:

![Screenshot 2024-01-19 at 15 54 25](https://github.com/WordPress/gutenberg/assets/1682452/671a5204-7450-4d8c-8c26-911ac01c08bd)

2
Consider to add an informational modal dialog as Slack does. 
In the Slack user interface, when repeatedly pressing the Tab key, a modal dialog opens with some useful information about keyboard usage. Screenshot:

![Screenshot 2024-01-19 at 09 43 43](https://github.com/WordPress/gutenberg/assets/1682452/85116d17-1d0e-487e-97e3-31be580a548f)

The dialog opens only when pressing the Tab key multiple times in a short amount of time. It's a sensible way to try to determine whether users are using the keyboard and provide them with useful information.

On one hand, a modal dialog that opens unexpectedly can be potentially disorienting. However, this dialog should open only once on first 'detection' and never open again. I'd tend to think it's something worth considering to welcome first time users who use the keyboard and provide them with important information.

Any thoughts and feedback very welcome.
Cc @joedolson @alexstine @andrewhayward

**Comment by alexstine** [comment]
@afercia Point 2, the Slack keyboard hint dialog only opens because Tab/Shift+Tab are no longer used for navigating regions/areas around the app. Tab/Shift+Tab navigate the content within a region and F6/Shift+F6 navigate regions. For example, if I am in the message compose box, Shift+Tab places me in the formatting toolbar and Shift+F6 places me in the left sidebar, channel/DMs list.

**Comment by joedolson** [comment]
Also re: point 2, I think there problems with a modal that launches without any clear information about how to find that information again. For any user with memory or cognitive issues, a single viewing doesn't provide a lot of helpful information. And first detection in an app is very different from first detection on a website. Most users will only use a single version of the app per device, but may use many different instances of WordPress. So they'd have to deal with that dialog repeatedly.

I think the first idea would be much simpler and more consistent.

**Comment by afercia** [comment]
Thanks all for your thoughts and feedback. Looks like the first point would be something nice to have, then.
